lib/goldiloader.rb

```diff
@@ -8,3 +8,37 @@
 require 'goldiloader/association_options'
 require 'goldiloader/association_loader'
 require 'goldiloader/active_record_patches'
+
+module Goldiloader
+  @globally_enabled = true
+
+  class << self
+
+    # Sets the process-wide enabled status
+    attr_reader :globally_enabled
```

Comment:
Should this be an `attr_accessor` to make this writable? 

```suggestion
    attr_accessor :globally_enabled
```

Also the defaulting on line 13 should use this setter (i.e. `self. globally_enabled = true`) to fail fast if we mistype the the variable name.